core/src/main/scala/org/apache/spark/sql/CachedDataFrame.scala

```diff
@@ -49,26 +52,38 @@ import org.apache.spark.unsafe.Platform
 import org.apache.spark.util.CallSite
 import org.apache.spark.{Logging, NarrowDependency, ShuffleDependency, SparkContext, SparkEnv, SparkException, TaskContext}
 
-class CachedDataFrame(df: Dataset[Row], val queryString: String,
+class CachedDataFrame(df: Dataset[Row], var queryString: String,
```

Comment:
This is because on the UI always the actual string should b edisplayed even if the cached plan is being used. The cached plan made could have been due to different set of constants but the current execution should reflect the actual current query executed.